engine/spverrors/unfold.go

```diff
@@ -0,0 +1,92 @@
+package spverrors
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/bitcoin-sv/spv-wallet/models"
+)
+
+type errListUnwrapper interface {
+	Unwrap() []error
+}
+
+// UnfoldError unfolds the error chain into a single string
+// example:
+//
+//	[errType1] outer error message -> [errType2] inner error message -> [errTypeN] innermost error message
+//
+// if error message is contained by outer error, it will be omitted, like:
+//
+//	[errType1] error message -> [errType2]
+//
+// if error is a joined error, it will be unfolded as:
+//
+//	[type] ([errType1] joined error1 AND [errType2] joined error2)
+//
+// for SPVError, it will print the status code in parentheses:
+//
+//	[models.SPVError(404)] error message
+func UnfoldError(err error) string {
+	if err == nil {
+		return ""
+	}
+
+	prevMsg := ""
+	result := strings.Builder{}
+	for current := err; current != nil; current = errors.Unwrap(current) {
+		msg := current.Error()
+
+		if prevMsg != "" {
+			result.WriteString(" -> ")
```

Comment:
maybe we could consider adding new lines :see_no_evil:

Reply:
zerolog would print them as `\n` and it's not - I think - you would like to get.
Besides - for production env - we'll most probably use json logger and I'm not sure if the new lines would ease the "reading" process.

Reply:
ok, for now, let's keep it as is